core/src/system_monitor_service.rs

```diff
@@ -728,25 +740,26 @@ impl SystemMonitorService {
                 edx: 0,
             };
 
-            let max_leaf = cpuid_mfr.eax;
-            let max_subleaf_7 = if 7 <= max_leaf {
-                __get_cpuid_max(7).1
-            } else {
-                0
+            let max_leaf = match CpuidParamValue::try_from(std::cmp::min(
+                cpuid_mfr.eax,
+                CPUID_PARAM_MAX_SUPPORTED_VALUE,
+            )) {
+                Ok(val) => val,
+                Err(_err) => CpuidParamValue::Manufacturer,
             };
 
             let mfr_id: i64 = if cpuid_mfr.ebx == CPUID_MANUFACTURER_EBX_INTEL
```

Comment:
consider making the `i64` here keep the `CpuManufacurer` type, then `i64::from(mfg_id)` once where the datapoint is built